src/reynir/bintokenizer.py

```diff
@@ -1228,24 +1245,27 @@ def given_names_or_middle_abbrev(
             ) -> Optional[List[PersonNameTuple]]:
                 """ Check for given name or middle abbreviation """
                 gnames = given_names(tok)
+                wrd = tok.txt
                 if gnames is not None:
-                    if tok.txt in BOTH_GIVEN_AND_FAMILY_NAMES:
+                    if wrd in BOTH_GIVEN_AND_FAMILY_NAMES:
                         # For instance "Hafstein" which can be both a given
                         # name and a family name: prepend the family name as
                         # an genderless and caseless option to the list
                         gnames = [
-                            PersonNameTuple(name=tok.txt, gender=None, case=None)
+                            PersonNameTuple(name=wrd, gender=None, case=None)
                         ] + gnames
                     return gnames
                 if tok.kind != TOK.WORD:
                     return None
-                wrd = tok.txt
-                if len(wrd) > 2 or wrd[0].islower():
-                    if wrd not in FOREIGN_MIDDLE_NAME_SET:
-                        # Accept "Thomas de Broglie", "Ruud van Nistelrooy"
-                        return None
-                # One or two letters, capitalized: accept as middle name abbrev,
-                # all genders and cases possible
+                if auto_uppercase and wrd in MIDDLE_NAME_ABBREVS:
+                    wrd = wrd.capitalize()
+                if len(wrd) > 3 or (
```

Comment:
Why the change to len > 3 from len > 2? A comment here would be good.

Reply:
Added commentary. The length was changed due to the following period being included in wrd if the abbreviation has a period. E.g. _th._ or _kr._ should be accepted as middle name abbreviations, but also shorter abbrevs such as _th_, _h._ or _b_